packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -117,6 +118,43 @@ exports.sourceNodes = async (
     pluginConfig,
   })
 
+  createTypes(`
+    interface ContentfulEntry @nodeInterface {
+      contentful_id: String!
+      id: ID!
+    }
+  `)
+  // This is not working:
+  // Error: Cannot find ObjectTypeComposer with name ContentfulAsset
```

Comment:
@vladar here I tried to have a generic ContentfulReference type. But I don't get it working with contentfulAsset. Any idea why?

Reply:
I think it's the other way around: `ContentfulReference` must be an interface while `ContentfulAsset` is an existing node type. So schema customization should look like this:

```js
createTypes(`
  interface ContentfulReference {
    contentful_id: String!
    id: ID!
  }
`)

schema.buildObjectType({
  name: `ContentfulAsset`,
  fields: {
    contentful_id: { type: `String!` },
    id: { type: `ID!` },
  },
  interfaces: [`ContentfulReference`, `Node`],
})

const gqlTypes = contentTypeItems.map(contentTypeItem =>
  schema.buildObjectType({
    name: _.upperFirst(_.camelCase(`Contentful ${contentTypeItem.name}`)),
    fields: {
      contentful_id: { type: `String!` },
      id: { type: `ID` },
    },
    interfaces: [`ContentfulReference`, `Node`],
  })
)
createTypes(gqlTypes)
```

Reply:
Thanks I'll try!